cmd/nodelink-controller/main.go

```diff
@@ -393,8 +393,22 @@ func (c *Controller) processNode(node *corev1.Node) error {
 		modNode.Labels[k] = v
 	}
 
-	// Taints are to be an authoritative list on the machine spec per cluster-api comments:
-	modNode.Spec.Taints = matchingMachine.Spec.Taints
+	// Taints are to be an authoritative list on the machine spec per cluster-api comments.
+	// However, we believe many components can directly taint a node and there is no direct source of truth that should enforce a single writer of taints
+	for _, mTaint := range matchingMachine.Spec.Taints {
```

Comment:
It'd be nice to create a function and unit test it

Reply:
sure.